cpp/src/reductions/mean.cu

```diff
@@ -27,6 +27,8 @@ std::unique_ptr<cudf::scalar> cudf::reduction::mean(column_view const& col,
                                                     rmm::mr::device_memory_resource* mr)
 {
   using reducer = cudf::reduction::compound::element_type_dispatcher<cudf::reduction::op::mean>;
+  auto col_type =
+    cudf::is_dictionary(col.type()) ? dictionary_column_view(col).keys().type() : col.type();
   return cudf::type_dispatcher(
-    col.type(), reducer(), col, output_dtype, /* ddof is not used for mean*/ 1, stream, mr);
+    col_type, reducer(), col, output_dtype, /* ddof is not used for mean*/ 1, stream, mr);
```

Comment:
rather than using `1`, better to have named variable with default value.

Reply:
I really don't know what this is.

Reply:
Ahh, I thought it was your change.